lib/ssh/src/ssh_connection_handler.erl

```diff
@@ -1259,15 +1267,31 @@ handle_event(info, {timeout, {_, From} = Request}, _,
 %%% Handle that ssh channels user process goes down
 handle_event(info, {'DOWN', _Ref, process, ChannelPid, _Reason}, _, D) ->
     Cache = cache(D),
-    ssh_client_channel:cache_foldl(
-      fun(#channel{user=U,
-                   local_id=Id}, Acc) when U == ChannelPid ->
-              ssh_client_channel:cache_delete(Cache, Id),
-              Acc;
+    %% Here we first collect the list of channel id's  handled by the process
+    %% Do NOT remove them from the cache - they are not closed yet!
+    Channels = ssh_client_channel:cache_foldl(
+      fun(#channel{user=U} = Channel, Acc) when U == ChannelPid ->
```

Comment:
previously this fun deleted the channel cache. now we are just sending channel close to peer.
I assume if peer responds with channel close we will delete entry from cache.

what if the peer does not reply? `idle_time` timeout will not handle that AFAIK.
what do you think about it? shouldn't they be removed after some timeout?

Reply:
You are right. Also the same can happen if our side requested channel to be closed but the peer failed to respond. I've prepared an update to handle such cases, will push after a bit of cleaning.

Reply:
Updated the code and some test, rebased to the latest 'maint'